### PR TITLE
[CELEBORN-171] Support JDK11.

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
To support JDK11.
I referred the spark patch 
https://issues.apache.org/jira/browse/SPARK-24421?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=16680169#comment-16680169

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/util/Utils.scala [hunk 1]
```diff
@@ -982,4 +983,35 @@ object Utils extends Logging {
...
+      bufferCleaner(buffer.asInstanceOf[DirectBuffer])
+    }
+  }
+
+  // In Java 8, the type of DirectBuffer.cleaner() was sun.misc.Cleaner, and it was possible
+  // to access the method sun.misc.Cleaner.clean() to invoke it. The type changed to
+  // jdk.internal.ref.Cleaner in later JDKs, and the .clean() method is not accessible even with
+  // reflection. However sun.misc.Unsafe added a invokeCleaner() method in JDK 9+ and this is
+  // still accessible with reflection.
+  private val bufferCleaner: DirectBuffer => Unit =
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/util/Utils.scala:998]: We can cache the cleanerMethod to avoid frequent reflection invocation

**Comment by waitinfuture** [comment]
IMO we'd better split this pr into two prs:
1. Refactor PartitionFileSorter
2. Support Java11

File: sbin/start-worker.sh [hunk 2]
```diff
@@ -32,10 +35,15 @@ if [ "$CELEBORN_WORKER_OFFHEAP_MEMORY" = "" ]; then
   CELEBORN_WORKER_OFFHEAP_MEMORY="1g"
 fi
 
-export CELEBORN_JAVA_OPTS="-Xmx$CELEBORN_WORKER_MEMORY -XX:MaxDirectMemorySize=$CELEBORN_WORKER_OFFHEAP_MEMORY $CELEBORN_WORKER_JAVA_OPTS"
+JAVA_VER=$(java -version 2>&1 | head -1 | cut -d'"' -f2 | sed '/^1\./s///' | cut -d'.' -f1)
+if [ "$JAVA_VER" -gt "8" ]; then
+  export CELEBORN_JAVA_OPTS="-Xmx$CELEBORN_WORKER_MEMORY -XX:MaxDirectMemorySize=$CELEBORN_WORKER_OFFHEAP_MEMORY $CELEBORN_WORKER_JAVA_OPTS --add-opens java.base/jdk.internal.misc=ALL-UNNAMED --illegal-access=warn -Dio.netty.tryReflectionSetAccessible=true"
+else
+  export CELEBORN_JAVA_OPTS="-Xmx$CELEBORN_WORKER_MEMORY -XX:MaxDirectMemorySize=$CELEBORN_WORKER_OFFHEAP_MEMORY $CELEBORN_WORKER_JAVA_OPTS"
+fi
```
Comment (pan3793) [inline sbin/start-worker.sh:43]:
```suggestion
export CELEBORN_JAVA_OPTS="-Xmx$CELEBORN_WORKER_MEMORY -XX:MaxDirectMemorySize=$CELEBORN_WORKER_OFFHEAP_MEMORY $CELEBORN_WORKER_JAVA_OPTS"
JAVA_VERSION=$(java -version 2>&1 | grep " version " | head -1 | awk '{print $3}' | tr -d '"')
if [[ ! "$JAVA_VERSION" == 1.8.* ]]; then
  export CELEBORN_JAVA_OPTS="${CELEBORN_JAVA_OPTS} --add-opens java.base/jdk.internal.misc=ALL-UNNAMED --illegal-access=warn -Dio.netty.tryReflectionSetAccessible=true"
fi
```
The above suggestion is based on:

When I set `JAVA_HOME=/path/jdk17` and run `java -version`, the first line of output is not the java version, so `grep " version "` is added, and I try my best to avoid using regex to make each shell command as short and simple enough.
```
➜  ~ java -version
[0.000s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /mnt/wsl/docker-desktop-bind-mounts/Ubuntu/85bae2ead9d613eb16150e89460f06daea72d31beaa1c37c68457a29a0fdba68/cpuset.
java version "17.0.4" 2022-07-19 LTS
Java(TM) SE Runtime Environment (build 17.0.4+11-LTS-179)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.4+11-LTS-179, mixed mode, sharing)
```

And the if clause is changed to avoid duplicated code.

File: pom.xml [hunk 3]
```diff
@@ -1011,6 +1011,20 @@
...
+        <java.version>8</java.version>
+      </properties>
+    </profile>
+
+    <profile>
+      <id>jdk-11</id>
+      <properties>
+        <java.version>11</java.version>
+      </properties>
+    </profile>
```
Comment (pan3793) [inline pom.xml:1026]:
```suggestion
    <profile>
      <id>jdk-11</id>
      <activation>
        <jdk>11</jdk>
      </activation>
      <properties>
        <java.version>11</java.version>
      </properties>
    </profile>
```

File: sbin/start-worker.sh [hunk 4]
```diff
@@ -19,7 +19,10 @@
 # Starts the celeborn worker on the machine this script is executed on.
 
 if [ -z "${CELEBORN_HOME}" ]; then
-  export CELEBORN_HOME="$(cd "`dirname "$0"`"/..; pwd)"
+  export CELEBORN_HOME="$(
+    cd "$(dirname "$0")"/..
+    pwd
+  )"
```
Comment (pan3793) [inline sbin/start-worker.sh:25]: unnecessary change

**Comment by FMX** [comment]
I'll close this pr and refactor it.
